src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests4.cs

```diff
@@ -713,27 +627,63 @@ public class C : ITuple
     public static void Main()
     {
         var t = new C();
-        Console.WriteLine(t is (3, 4)); // false
-        Console.WriteLine(t is (3, 4, 5)); // TRUE
+        Console.WriteLine(t is (7, 8)); // true (Extensions.Deconstruct)
+        Console.WriteLine(t is (3, 4, 5)); // true via ITuple
         Console.WriteLine(t is (3, 0, 5)); // false
         Console.WriteLine(t is (3, 4, 5, 6)); // false
     }
 }
 static class Extensions
 {
-    public static void Deconstruct(this C c, out int X, out int Y) => (X, Y) = (3, 4);
+    public static void Deconstruct(this C c, out int X, out int Y) => (X, Y) = (7, 8);
 }
 ";
             var compilation = CreatePatternCompilation(source);
             compilation.VerifyDiagnostics();
             var expectedOutput =
-@"False
+@"True
 True
 False
 False";
             var compVerifier = CompileAndVerify(compilation, expectedOutput: expectedOutput);
         }
 
+        [Fact]
+        public void ITuple_09b()
+        {
+            // - An extension Deconstruct hides ITuple
+            var source =
+@"using System;
+using System.Runtime.CompilerServices;
+public class C : ITuple
+{
+    int ITuple.Length => 4;
```

Comment:
>4; [](start = 25, length = 2)

Consider splitting this test. One test would have the ITuple implementation throw.
Also, three of the comments say the same thing but in different ways...